### ISSUE TITLE
textInput and other components inside InputAccessoryView do not show on initial rendering 

### ISSUE THREAD

**Issue by Joshua-Zhang-S** [facebook/react-native#31778]
## Description
All components (including textInput) inside InputAccessoryView do not show initially. Only when you input anything, it will show. After navigating back to this page again later, the keyboard will not show. here is the code inside InputAccessoryView:

<InputAccessoryView backgroundColor="#fffffff7">
                    <View style={styles.searchSection} >
                        <Icon style={styles.searchIcon} name='search' size={20} color='#000' />
                        <TextInput
                            ref={ele => this.inputRef = ele}
                            style={styles.textInput}
                            placeholder="Search BAYA ... "
                            placeholderTextColor='grey'
                            autoFocus={true}
                            onChangeText={text => this._searchItems(text)}
                            clearButtonMode='always'
                            onSubmitEditing={text => this._submitAndClear}
                            blurOnSubmit={true}
                        />
                    <Button
          onPress={
            this._submitAndClear
          }
          title="Search"
        />
                    </View>
                </InputAccessoryView>
## React Native version:
Run `react-native info` in your terminal and copy the results here.
System:
    OS: macOS 11.4
    CPU: (8) x64 Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
    Memory: 1.52 GB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.17.0 - /usr/local/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.10.1 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.5, DriverKit 20.4, macOS 11.3, tvOS 14.5, watchOS 7.4
    Android SDK:
      API Levels: 28, 29, 30
      Build Tools: 30.0.3
      System Images: android-28 | Intel x86 Atom_64, android-29 | Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: Not Found
    Xcode: 12.5/12E262 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_291 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1 
    react-native: 0.64.2 => 0.64.2 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

export default SearchProduct
1. After navigation into this page, the search section doesn't show.
2. Only after typing anything, the search section will show.
3. After navigating back to this page, the keyboard doesn't show.

## Expected Results
The keyboard and searchSection should be always visible on this page.

## Snack, code example, screenshot, or link to a repository:
Here is the code:
import React from 'react';
import {
    StyleSheet, View, TextInput, Pressable, FlatList, Text, Button, InputAccessoryView
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

import searchTerms from '../data/searchTerms';

class SearchProduct extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            data: [],
            inputText: 'Search Baya',
        };
    };

    //Dummy data
    _renderSeparator = () => {
        return (
            <View
                style={{
                    height: 1,
                    width: '100%',
                    backgroundColor: '#CED0CE',
                }}
            />
        );
    };

    _searchItems = text => {
        if (text.length <= 0) { return; }
        let newData = searchTerms.filter(item => {
            const itemData = `${item.key.toUpperCase()}`;
            const textData = text.toUpperCase();
            return itemData.startsWith(textData);
        });
        this.setState({
            data: newData,
            inputText: text
        });
        console.log("Inputed: " + text + " but state is: " + this.state.inputText);
        //        console.log(this.state.data);

    };

    _navigation = (text) => {
        //        console.log('search product: ' + text);
        this.props.navigation.navigate('ProductRelated', { name: text });
    }

    _onPressSearchedText = (name) => {
        //        console.warn('Item Pressed ' + name);
        this.setState({
            inputText: name
        });
        //        console.log("Presseed: " + name + " Input state is: " + this.state.inputText);
        this._navigation(name);
    };

    _submitAndClear = () => {
        console.log('In SearchProduct submit clear. state:' + this.state.inputText)
        this._navigation(this.state.inputText);
    };

    componentDidMount() {
        //this.inputRef.focus();
        this.setState({
            inputText: ''
        });
        console.log('componentDidMount ' + this.state.inputText);
        if (this.inputRef.current) {
            setTimeout(() => this.inputRef.current.focus(), 500);
        }

    };

    render() {
        return (
            <View>
                <FlatList
                    data={this.state.data}
                    renderItem={({ item }) => (
                        <Pressable onPress={() =>
                            this._onPressSearchedText(item.key)}>
                            <Text style={styles.searchedText}>{item.key} </Text>
                        </Pressable>
                    )}
                    keyExtractor={(item, index) => item.key}
                    ItemSeparatorComponent={this._renderSeparator}
                    keyboardShouldPersistTaps={'handled'}
                />
                <InputAccessoryView backgroundColor="#fffffff7">
                    <View style={styles.searchSection} >
                        <Icon style={styles.searchIcon} name='search' size={20} color='#000' />
                        <TextInput
                            ref={ele => this.inputRef = ele}
                            style={styles.textInput}
                            placeholder="Search BAYA ... "
                            placeholderTextColor='grey'
                            autoFocus={true}
                            onChangeText={text => this._searchItems(text)}
                            clearButtonMode='always'
                            onSubmitEditing={text => this._submitAndClear}
                            blurOnSubmit={true}
                        />
                        <Button
                            onPress={
                                this._submitAndClear
                            }
                            title="Search"
                        />
                    </View>
                </InputAccessoryView>
            </View>
        );
    };

}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    searchSection: {
        flexDirection: 'row',
        backgroundColor: '#fff',
        borderColor: 'grey',
        borderWidth: 2,
        height: 40,
        borderRadius: 5,
        flex: 1,
    },
    searchIcon: {
        margin: 5,
        alignItems: 'center',
    },
    textInput: {
        height: 30,
        flex: 1,
    },
    searchedText: {
        fontSize: 25,
        padding: 10,
    },
})

**Comment by cortinico** [comment]
Hey,
We're cleaning up the React Native issue tracker to make sure we focus on the most important issues first.

I'm closing this one because it's stale and related to a version of React Native we no longer support. If the issue persists on the latest version, please re-open a new issue and reference this one.

Thank you.

**Comment by fabriziobertoglio1987** [comment]
Do you still experience this issue? 

I have four years of experience maintaining facebook/react-native and I specialize in the Text and TextInput components. I currently have 58 [facebook/react-native PRs](https://github.com/facebook/react-native/pulls?q=is%3Apr+author%3Afabriziobertoglio1987+). 

If you still experience this issue, I will prepare a patched release with the fix.

Thanks a lot

**Comment by alextbogdanov** [comment]
I'm having this issue in react-native 0.76.9 with new arch enabled

**Comment by alextbogdanov** [comment]
Found a fix! Just wrap the InputAccessoryView in a <View>
